applications/ContactStructuralMechanicsApplication/custom_utilities/interface_preprocess.cpp

```diff
@@ -119,12 +127,49 @@ void InterfacePreprocessCondition::GenerateInterfacePart<3>(
 /***********************************************************************************/
 /***********************************************************************************/
 
+std::unordered_map<IndexType, Properties::Pointer> InterfacePreprocessCondition::CreateNewProperties(ModelPart& rOriginPart)
+{
+    const SizeType number_properties = mrMainModelPart.NumberOfProperties();
+
+    std::unordered_map<IndexType, Properties::Pointer> new_properties;
+    // Reorder id Properties
+    IndexType count = 0;
+    for (auto it_prop = mrMainModelPart.PropertiesBegin(); it_prop < mrMainModelPart.PropertiesEnd(); ++it_prop) {
+        ++count;
+        it_prop->SetId(count);
+    }
+    // Create new properties
+    count = 0;
+    for (auto it_prop = rOriginPart.PropertiesBegin(); it_prop < rOriginPart.PropertiesEnd(); ++it_prop) {
+        Properties::Pointer p_original_prop = *(it_prop.base());
+        ++count;
+        Properties new_prop(number_properties + count + 1);
+        Properties::Pointer p_new_prop = Kratos::make_shared<Properties>(new_prop);
```

Comment:
i think there is a ModelPart::AddProerty for this or something similar.
[edit:] you are doing exactly what i was afraid of...creating without passing through te modelpart interface

Reply:
use model_part.GetProperties( id )

if it is not there, it will create it new

Reply:
I am using the interface, look at my last commit

Reply:
Using what you tell me to use gives a memory error